### PR TITLE
Update sepolicy for audio hidl one image

### PR DESCRIPTION
**Pull request by haochuan1982**
Update sepolicy for audio hidl one image

Tracked-On: OAM-128617

### CODE REVIEW

File: audio/file_contexts [hunk 1]
```diff
@@ -0,0 +1 @@
+/vendor/bin/set_board_prop.sh u:object_r:set_board_prop_exec:s0
```
Comment (ceiba1985) [inline audio/file_contexts:1]: Please ensure you are using "#!/vendor/bin/sh"

File: audio/hal_audio_default.te [hunk 2]
```diff
@@ -1 +1,2 @@
 get_prop(hal_audio_default, vendor_audio_default_prop)
+get_prop(hal_audio_default, audio_product_board_prop)
```
Comment (ceiba1985) [inline audio/hal_audio_default.te]: add vendor prefix "vendor_audio_product_board_prop"

File: audio/property_contexts [hunk 3]
```diff
@@ -1,2 +1,5 @@
 ro.vendor.hdmi.audio    u:object_r:vendor_audio_default_prop:s0
 vendor.intel.enable.audio u:object_r:vendor_audio_default_prop:s0
+audio.product.board_type  u:object_r:audio_product_board_prop:s0
```
Comment (ceiba1985) [inline audio/property_contexts]: use vendor.audio.product.board_type instead.

File: audio/set_board_prop.te [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type set_board_prop, domain;
+type set_board_prop_exec, exec_type, file_type, vendor_file_type;
+init_daemon_domain(set_board_prop)
+
+set_prop(set_board_prop, audio_product_board_prop)
+
+allow set_board_prop vendor_toolbox_exec:file execute_no_trans;
```
Comment (ceiba1985) [inline audio/set_board_prop.te]: can remove this line

File: audio/set_board_prop.te [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+allow set_board_prop vendor_shell_exec:file rx_file_perms;
+allow set_board_prop vendor_toolbox_exec:file rx_file_perms;
+allow set_board_prop shell_exec:file rx_file_perms;
```
Comment (ceiba1985) [inline audio/set_board_prop.te]: remove this line, under treble, it's not proper for a vendor domain to access a core domain.

File: audio/shell.te [hunk 6]
```diff
@@ -0,0 +1 @@
+allow shell set_board_prop_exec:file getattr;
```
Comment (ceiba1985) [inline audio/shell.te]: this also should not be needed

File: audio/file_contexts (hunk 1 above)
Comment (haochuan1982) [reply audio/file_contexts:1]: thanks, script already ensured
https://github.com/projectceladon/device-androidia-mixins/pull/2191/files#diff-c2869d63e6d7b2d925bd61e8a4f791ae599064af30868e98374fdc73f8c07b22

File: audio/hal_audio_default.te (hunk 2 above)
Comment (haochuan1982) [reply audio/hal_audio_default.te]: updated

File: audio/set_board_prop.te (hunk 4 above)
Comment (haochuan1982) [reply audio/set_board_prop.te]: updated

File: audio/property_contexts (hunk 3 above)
Comment (haochuan1982) [reply audio/property_contexts]: updated

File: audio/shell.te (hunk 6 above)
Comment (haochuan1982) [reply audio/shell.te]: updated

File: audio/set_board_prop.te (hunk 5 above)
Comment (haochuan1982) [reply audio/set_board_prop.te]: updated

**Review by ceiba1985 — APPROVED** [review]

**Review by dyang23 — APPROVED** [review]

**Review by dyang23 — APPROVED** [review]

**Pull request merged**